### PR TITLE
bugsbash: Use net.JoinHostPort

### PR DESCRIPTION
**Pull request by soniasingla**

### CODE REVIEW

**Comment by haircommander** [comment]
can you drop the changes to the vendor directory? https://github.com/cri-o/cri-o/pull/4864 will make them not necessary

File: vendor/github.com/go-git/go-git/v5/plumbing/transport/git/common.go [hunk 1]
```diff
@@ -92,7 +93,7 @@ func (c *command) StdoutPipe() (io.Reader, error) {
 func endpointToCommand(cmd string, ep *transport.Endpoint) string {
 	host := ep.Host
 	if ep.Port != DefaultPort {
-		host = fmt.Sprintf("%s:%d", ep.Host, ep.Port)
+		host = net.JoinHostPort(ep.Host, strconv.Itoa(ep.Port))
```
Comment (soniasingla) [inline vendor/github.com/go-git/go-git/v5/plumbing/transport/git/common.go:96]: @haircommander can you help if i used correct conv here and in other cases

Comment (haircommander) [reply vendor/github.com/go-git/go-git/v5/plumbing/transport/git/common.go:96]: in general it looks good to me. though this change and every other in the vendor directory should be dropped :)

**Pull request closed without merging**